### PR TITLE
Button updates

### LINKED ISSUE #1864: Button boxshadow should reduce when depressed
acceptance criteria
- when i hover a button, the button box shadow goes up a level or 2
- when i click a button, the button shadow should go down a level or 2 as the button goes down, otherwise it visually looks like the button is changing on the y axis instead of the z axis

effort: 10.5 points

### PR DESCRIPTION
**Pull request by ackernaut**
Fixes #1864 

Also, I realized that `<PrimaryButton>` wasn’t giving us additional functionality besides being scoped to the primary gradient color, so I’m yanking in favor of only making updates on `<Button>`. I realize Mr. Button is a large file, but I’d rather only have to focus on that 1 file and use `buttonStyle="primary"` for those buttons.

### CODE REVIEW

**Comment by ackernaut** [comment]
Once again, any volunteers to review this one? The buttons go up a level of depth on hover/focus/active, and then go down a level of depth when pressed. Depth is nuked when disabled/waiting. Yanked PrimaryButton in favor of only having to keep up with Button. @mattkrick @dan-f

File: src/universal/components/ResetPasswordPage/ResetPasswordForm.js [hunk 1]
```diff
@@ -44,9 +44,14 @@ const PasswordResetForm = (props: Props) => {
           disabled={props.submitting}
         />
       </Block>
-      <PrimaryButton disabled={!props.valid} waiting={props.submitting} type="submit">
```
Comment (mattkrick) [inline src/universal/components/ResetPasswordPage/ResetPasswordForm.js]: I'm generally in favor of more components with fewer props to reduce component complexity (and the button component has a LOT of complexity!)

In the future, I'd love to break it down & start with a `PlainButton` that does nothing but remove html styles.
Then, we could build a RaisedButton on top of that, which adds hover & click transitions + a few other styles.
Then, we could have different styles of a RaisedButton that fill in the props for us.

That said, this LGTM. Personally, i still think it's weird that none of the buttons have box shadows, but maybe that's just me.

Comment (dan-f) [reply src/universal/components/ResetPasswordPage/ResetPasswordForm.js]: 👏 Yes to more components w/ fewer props! I generally also prefer things to be compositionally  over parametrically dynamic.

Comment (ackernaut) [reply src/universal/components/ResetPasswordPage/ResetPasswordForm.js]: I agree in that direction. In this case I was having to work on styles for state in 2 places, so it wasn’t really broken down in a way that is progressively more enhanced, it was doing the same thing (there were 2 ways to have a primary button and inconsistent styles between the 2).

I would be glad to capture an issue to refactor button variants.

**Comment by dan-f** [comment]
🔥

**Pull request merged**